File: fizzbuzz.lua
function fizzbuzz(i)
  local result = ""
  if i % 3 == 0 then result = result .. "Fizz" end
  if i % 5 == 0 then result = result .. "Buzz" end
  return result ~= "" and result or tostring(i)
end

for i = 1, 100 do
  print(fizzbuzz(i))
end

File: fizzbuzz.java
public class Main {
  public static void main (String[] args) {
    for (int i = 1; i <= 100; i++) {
      System.out.println(fizzBuzz(i));
    }
  }

  public static String fizzBuzz(int i) {
    StringBuilder result = new StringBuilder();
    if (i % 3 == 0) result.append("Fizz");
    if (i % 5 == 0) result.append("Buzz");
    return result.length() > 0 ? result.toString() : String.valueOf(i);
  }
}
